### PR TITLE
sys-devel/llvm: a few feature enhancements

### PR DESCRIPTION
**Pull request by zzlei**
- add support for building llvm against musl
- enable clang to build binaries against musl
- introduce USE flag "sanitizer" to control the building of compiler-rt's
  sanitizers (they cause problem on musl)
- be able to override default values of -stdlib and -rtlib for clang
  * USE="libcxx" implies -stdlib=lib++ (originally libstdc++)
  * USE="compiler-rt" implies -rtlib=compiler-rt (originally libgcc)

### CODE REVIEW

**Comment by doughdemon** [comment]
Consider renaming "sanitizer" to "sanitize" as used for gcc.

The musl support is only x86 and amd64?

**Comment by zzlei** [comment]
> The musl support is only x86 and amd64?

I intentionally left out patches for ARM in this PR, since they are relatively HUGE.

Besides that, I think it should work on POWER and MIPS as is. I've never tested it on non-x86 platforms though.

**Comment by mgorny** [comment]
What's upstreaming status of those patches?

File: sys-devel/llvm/metadata.xml [hunk 1]
```diff
@@ -26,5 +26,8 @@
 		<flag name="multitarget">Build all host targets (default: host only)</flag>
 		<flag name="ncurses">Support querying terminal properties using ncurses' terminfo</flag>
 		<flag name="static-analyzer">Install the Clang static analyzer (requires USE=clang)</flag>
+		<flag name="sanitize">Build compiler-rt's sanitizers</flag>
+		<flag name="libcxx">Use libc++ instead of libstdc++ as the default stdlib for clang</flag>
+		<flag name="compiler-rt">Use compiler-rt instead of libgcc as the default rtlib for clang</flag>
 	</use>
```
Comment (mgorny) [inline sys-devel/llvm/metadata.xml]: Please order flags lexically.

**Comment by zzlei** [comment]
> What's upstreaming status of those patches?

- Support for musl (including ARM support) is already in the trunk; the fix for building LLVM against musl isn't, and is unlikely to be accepted as-is.
- Setting default value for -stdlib is supported, but not for -rtlib. I'm trying to augment that.

**Comment by zzlei** [comment]
I'm closing this PR to combine it with my other related PRs into a new one, for the sake of easier reviewing.

**Pull request closed without merging**